src/main/webapp/js/feedbackResponseComments.js

```diff
@@ -119,55 +119,58 @@ var editCommentHandler = function(e) {
-                    setFormErrorMessage(editForm.find('div > a'), data.errorMessage);
-                } else if (frCommentList.parent().find('div.delete_error_msg').length === 0) {
-                    frCommentList.after('<div class="delete_error_msg alert alert-danger">' + data.errorMessage + '</div>');
-                }
-                submitButton.html('<span class="glyphicon glyphicon-trash glyphicon-primary"></span>');
-            } else if (isInCommentsPage()) {
-                reloadFeedbackResponseComments(formObject, panelHeading);
-            } else {
-                var numberOfItemInFrCommentList = deletedCommentRow.parent().children('li');
-                if (numberOfItemInFrCommentList.length <= 2) {
-                    deletedCommentRow.parent().hide();
-                }
-                if (frCommentList.find('li').length <= 1) {
-                    frCommentList.hide();
+
+    BootboxWrapper.showModalConfirmation('Warning', 'Are you sure you want to remove this comment?', function() {
+        var formObject = submitButton.parent();
+        var deletedCommentRow = submitButton.closest('li');
+        var formData = formObject.serialize();
+        var frCommentList = submitButton.closest('.comments');
+        var panelHeading = submitButton.parents("[id^='panel_display-']").find('.panel-heading').first();
+
+        $.ajax({
+            type: 'POST',
+            url: submitButton.attr('href') + '?' + formData,
+            beforeSend: function() {
+                submitButton.html("<img src='/images/ajax-loader.gif'/>");
+            },
+            error: function() {
+                processDeleteError('Failed to delete comment. Please try again.', submitButton);
+            },
+            success: function(data) {
+                if (data.isError) {
+                    processDeleteError(data.errorMessage, submitButton);
+                } else if (isInCommentsPage()) {
+                    reloadFeedbackResponseComments(formObject, panelHeading);
+                } else {
+                    var numberOfItemInFrCommentList = deletedCommentRow.parent().children('li');
+                    if (numberOfItemInFrCommentList.length <= 2) {
+                        deletedCommentRow.parent().hide();
+                    }
+                    if (frCommentList.find('li').length <= 1) {
+                        frCommentList.hide();
+                    }
+                    deletedCommentRow.remove();
+                    frCommentList.parent().find('div.delete_error_msg').remove();
                 }
-                deletedCommentRow.remove();
-                frCommentList.parent().find('div.delete_error_msg').remove();
             }
-        }
-    });
+        });
+    }, null, BootboxWrapper.DEFAULT_OK_TEXT, BootboxWrapper.DEFAULT_CANCEL_TEXT, StatusType.WARNING);
 };
 
+function processDeleteError(errorMessage, submitButton) {
+    var editForm = submitButton.parent().next().next().next();
```

Comment:
You're right, but the thing is that currently I can't realize what cssSelector is expected for this editForm. Moreover, all this refactoring is outside of the main task of this issue